core/base/src/TSystem.cxx

```diff
@@ -1146,7 +1146,7 @@ Bool_t TSystem::ExpandFileName(const char *fname, char *xname, const int kBufSiz
    } else if (c[0] == '~' && c[1] != '/') { // ~user case
       n = strcspn(c+1, "/ ");
       buff[0] = 0;
-      strncat(buff, c+1, n);
+      (void) strlcat(buff, std::string(c+1).c_str(), sizeof(buff) - 1);
```

Comment:
This change the semantic of the code.  It needs to stop copying the string at 'n'.  so more like
```
strlcat(buf, c+1, std::min(n, sizeof buff - 1);
```

Reply:
Yes, I remember that... I thought you had fixed that one. I will change.